[CMSP-12] Multisite search-replace EA (#8303)
pantheon-systems/documentation: source/content/guides/multisite/07-search-replace.md

---
title: WordPress Multisite

> subtitle: Search-Replace

Author: ```suggestion
subtitle: Search & Replace
```

> description: Additional handing required for running search-replace on WordPress Multisite.

Author: ```suggestion
description: Additional handing required for running search & replace on WordPress Multisites.
```

type: guide
getfeedbackform: default
contenttype: [guide]
innav: [false]
categories: [cms, troubleshooting]
cms: [wordpress]
audience: [development]
product: [--]
integration: [--]
tags: [multisite]
permalink: docs/guides/multisite/search-replace/
editpath: multisite/07-search-replace.md
---

<Alert title="Early Access" type="info" icon="leaf">

> WordPress Multisite (WPMS) Search-Replace is available for [Early Access](/oss-support-levels#early-access) participants. Features for WPMS Search-Replace are in active development. Pantheon's development team is rolling out new functionality often while this product is in Early Access. Visit [#wordpress in our community Slack](https://pantheon-community.slack.com/archives/CT8MC5Y0K) (you can sign up for the [Pantheon Slack channel here](https://slackin.pantheon.io/) if you don't already have an account) to learn how you can enroll in our Early Access program. Please review Pantheon's [Software Evaluation Licensing Terms](https://legal.pantheon.io/#contract-hkqlbwpxo) for more information about access to our software.

Reviewer: Search & Replace

Author: ```suggestion
WordPress Multisite (WPMS) search & replace is available for [Early Access](/oss-support-levels#early-access) participants. Features for WPMS search & replace are in active development. Pantheon's development team is rolling out new functionality often while this product is in Early Access. Visit [#wordpress in our community Slack](https://pantheon-community.slack.com/archives/CT8MC5Y0K) (you can sign up for the [Pantheon Slack channel here](https://slackin.pantheon.io/) if you don't already have an account) to learn how you can enroll in our Early Access program. Please review Pantheon's [Software Evaluation Licensing Terms](https://legal.pantheon.io/#contract-hkqlbwpxo) for more information about access to our software.
```

</Alert>

> This section provides information on how to configure automatic platform search-replace of URLs for both subdirectory and subdomain WordPress multisites.

Author: ```suggestion
This section provides information on how to configure automatic platform search-replace of URLs for both subdirectory and subdomain WordPress Multisites.
```

## Requirements
A WordPress Multisite

To enable search replace between environments on a WPMS site, add 
```yaml:title=pantheon.yml
search_replace: true
```
to [pantheon.yml](/pantheon-yml).

Note, if pantheon.yml is different between environments, the search_replace value in the source environment’s pantheon.yml will determine if the job will be run or not.

### Subdomain WPMS
For subdomain multisites, environments to be replaced are defined and paired in the `sites.yml` file. Currently, the file must be committed at `private/sites.yml` in the site’s git repository.

See [this gist](https://gist.github.com/scottbuscemi/b051ad6510ef8494aff80d0f43afeeb2) for an example of how sites.yml might look.

> Search-replace will run for each domain listed in the source environment that has a matching key in the target environment.

Author: @jazzsequence, copying your comment here:
> in that case I would make sure there’s adequate info in a gh repo somewhere that the docs can link out to because there’s some nuance about how it works that should be explained more fully imo

There won't be a github repo for now, what nuance would you like clarified here?

Reviewer 2: Discussion [here](https://pantheon.slack.com/archives/C0397K11QEQ/p1680628327865289).

tl:dr; Broadly I'd like a description of the `sites.yml` file linked as well as information and troubleshooting steps about what validation errors might trigger and how to solve them linked in here somewhere, in addition to an example valid and annotated `sites.yml` file.

Related https://github.com/pantheon-systems/pyml-validator/issues/13

Author: Will address as part of CMSP-319 and the docs updates to come out of that.

If search-replace is enabled for an environment, but `sites.yml` does not exist, nothing will be updated. if `sites.yml` is different between environments, the domain_maps in the target environment’s `sites.yml` will be used to determine what to replace.

> There is currently a limit of twenty-five sites that can be configured for search-replace. We expect this number to increase before general availability.

Reviewer: 25

Author: ```suggestion
There is currently a limit of 25 sites that can be configured for search-replace. We expect this number to increase before general availability.
```

---
title: WordPress Multisite
subtitle: Search & Replace
description: Additional handing required for running search & replace on WordPress Multisites.
type: guide
getfeedbackform: default
contenttype: [guide]
innav: [false]
categories: [cms, troubleshooting]
cms: [wordpress]
audience: [development]
product: [--]
integration: [--]
tags: [multisite]
permalink: docs/guides/multisite/search-replace/
editpath: multisite/07-search-replace.md
---

<Alert title="Early Access" type="info" icon="leaf">

> WordPress Multisite (WPMS) search & replace is available for [Early Access](/oss-support-levels#early-access) participants. Features for WPMS search & replace are in active development. Pantheon's development team is rolling out new functionality often while this product is in Early Access. Visit [#wordpress in our community Slack](https://pantheon-community.slack.com/archives/CT8MC5Y0K) (you can sign up for the [Pantheon Slack channel here](https://slackin.pantheon.io/) if you don't already have an account) to learn how you can enroll in our Early Access program. Please review Pantheon's [Software Evaluation Licensing Terms](https://legal.pantheon.io/#contract-hkqlbwpxo) for more information about access to our software.

Reviewer 3: ```suggestion
WordPress Multisite (WPMS) search & replace is available for [Early Access](/oss-support-levels#early-access) participants. Features for platform search-replace are in active development. Pantheon's development team is rolling out new functionality often while this product is in Early Access. Visit [#wordpress in our community Slack](https://pantheon-community.slack.com/archives/CT8MC5Y0K) (you can sign up for the [Pantheon Slack channel here](https://slackin.pantheon.io/) if you don't already have an account) to learn how you can enroll in our Early Access program. Please review Pantheon's [Software Evaluation Licensing Terms](https://legal.pantheon.io/#contract-hkqlbwpxo) for more information about access to our software.
```

This is referred to as `platform search-replace` further down in this document - we should decide on `WPMS search-replace` or `platform search-replace` and be consistent throughout the doc.

Author: re: WPMS search & replace vs platform search & replace— trying to differentiate as this behaves somewhat differently than _any s&r_ on the platform, and you can't *opt-in/out* of s&r on a non-WPMS site so that seemed confusing.

</Alert>

This section provides information on how to configure automatic platform search-replace of URLs for both subdirectory and subdomain WordPress Multisites.

## Requirements
A [WordPress Multisite](/guides/multisite)

## Opting In

> To enable search replace between environments on a WPMS site, add 

Author: search-replace was what I started with, but search & replace was @scottbuscemi preference in a previous review; I was not comprehensive enough in my followup throughout the document.

> To enable search replace between environments on a WPMS site, add 

Author: actually I see `search and replace` in docs I didn't touch, going to followup with that 😁

> To enable search replace between environments on a WPMS site, add 

Reviewer 4: This could be `and` or `&`, but I think it needs one of them.
```suggestion
To enable search and replace between environments on a WPMS site, add 
```

```yaml:title=pantheon.yml
search_replace: true
```
to [pantheon.yml](/pantheon-yml).

Note, if `pantheon.yml` is different between environments, the search_replace value in the source environment’s `pantheon.yml` will determine if the job will be run or not.

Search & replace runs on the platform as part of creating an environment, deploying to test or live, and when cloning content between environments.

### Subdomain WPMS
For subdomain Multisites, environments to be replaced are defined and paired in the `sites.yml` file. Currently, the file must be committed at `private/sites.yml` in the site’s git repository.

See [this gist](https://gist.github.com/scottbuscemi/b051ad6510ef8494aff80d0f43afeeb2) for an example of how `sites.yml` might look.

Search-replace will run for each domain listed in the source environment that has a matching key in the target environment.

> If search-replace is enabled for an environment, but `sites.yml` does not exist, nothing will be updated. if `sites.yml` is different between environments, the domain_maps in the target environment’s `sites.yml` will be used to determine what to replace.

Reviewer 3: ```suggestion
If search-replace is enabled for an environment, but `sites.yml` does not exist, nothing will be updated. if `sites.yml` is different between environments, the `domain_maps` in the target environment’s `sites.yml` will be used to determine what to replace.
```

> If search-replace is enabled for an environment, but `sites.yml` does not exist, nothing will be updated. if `sites.yml` is different between environments, the domain_maps in the target environment’s `sites.yml` will be used to determine what to replace.

Author: ```suggestion
If search-replace is enabled for an environment, but `sites.yml` does not exist, nothing will be updated. if `sites.yml` is different between environments, the `domain_maps` in the target environment’s `sites.yml` will be used to determine what to replace.
```

> If search and replace is enabled for an environment, but `sites.yml` does not exist, nothing will be updated. if `sites.yml` is different between environments, the `domain_maps` in the target environment’s `sites.yml` will be used to determine what to replace.

Reviewer 4: ```suggestion
If search and replace is enabled for an environment, but `sites.yml` does not exist, nothing will be updated. If `sites.yml` is different between environments, the `domain_maps` in the target environment’s `sites.yml` will be used to determine what to replace.
```

> There is currently a limit of 25 sites that can be configured for search and replace. We expect this number to increase before general availability.

Reviewer 4: ```suggestion
There is currently a limit of 25 sites that can be configured for search and replace. We expect this number to increase before General Availability.
```
